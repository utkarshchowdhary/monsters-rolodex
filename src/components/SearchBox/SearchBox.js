import React from 'react';

import './SearchBox.scss';

const SearchBox = ({ placeholder, handleTextChange }) => {
  return (
    <div>
      <input
        className="search"
        type="search"
        placeholder={placeholder}
        onChange={handleTextChange}
      />
    </div>
  );
};

export { SearchBox as default };
