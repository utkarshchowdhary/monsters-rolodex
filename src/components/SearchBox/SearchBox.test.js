import React from 'react';
import { shallow } from 'enzyme';
import SearchBox from './SearchBox';

let mockPlaceholder, mockHandleTextChange, wrapper;

beforeEach(() => {
  mockPlaceholder = 'Some Content';
  mockHandleTextChange = jest.fn();
  wrapper = shallow(
    <SearchBox
      placeholder={mockPlaceholder}
      handleTextChange={mockHandleTextChange}
    />
  );
});

it('should render SearchBox component', () => {
  expect(wrapper).toMatchSnapshot();
});

it('should call handleTextChange on input change', () => {
  wrapper.find('input').at(0).simulate('change');
  expect(mockHandleTextChange).toHaveBeenCalled();
});
