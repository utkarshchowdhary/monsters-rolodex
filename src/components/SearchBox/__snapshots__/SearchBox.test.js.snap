// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SearchBox component 1`] = `ShallowWrapper {}`;
